### PR TITLE
Reorganize "core" and "source" modules.

### PR DESCRIPTION
**Pull request by troessner**
Implements #438

### CODE REVIEW

File: lib/reek/core/examiner.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
+require_relative '../source/source_repository'
+require_relative 'warning_collector'
+require_relative 'smell_repository'
+require_relative 'tree_walker'
+
+module Reek
+  module Core
```
Comment (troessner) [inline lib/reek/core/examiner.rb]: Now Examiner is scoped under Core so everything is indented two to the right, hence the diff. Same goes for SmellWarning below.

**Comment by chastell** [comment]
:+1:, merging!

File: lib/reek/smells/smells.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require 'require_all'
 
-require_rel 'smells'
+require_rel '.'
```
Comment (mvz) [inline lib/reek/smells/smells.rb]: I'm rather surprised that this works.

Comment (mvz) [inline lib/reek/smells/smells.rb]: Apparently, it doesn't: #453.

File: lib/reek/spec/spec.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
-require_relative 'spec/should_reek'
-require_relative 'spec/should_reek_of'
-require_relative 'spec/should_reek_only_of'
+require_relative 'should_reek'
+require_relative 'should_reek_of'
+require_relative 'should_reek_only_of'
```
Comment (mvz) [inline lib/reek/spec/spec.rb]: I would prefer to keep this file at `reek/spec.rb`, for two reasons:

* When a namespace *foo* is introduced, I prefer the idiom of having `foo.rb` and `foo/some_class.rb` over `foo/foo.rb` or `foo/base.rb`.
* Users will now have to `require 'reek/spec/spec'`, which is a breaking change and looks odd.

Comment (troessner) [inline lib/reek/spec/spec.rb]: Yes, makes sense.

Comment (troessner) [inline lib/reek/spec/spec.rb]: @mvz please check #454

**Comment by chastell** [comment]
Argh, apologies for the trigger-happy merge. Thanks for fixing the warts in separate PRs.

**Pull request merged**